communication/nats_discovery/address.go

```diff
@@ -17,8 +18,12 @@ func NewAddress(server string, port int, topic string) *NatsAddress {
 	}
 }
 
-func NewAddressForIdentity(identity identity.Identity) *NatsAddress {
-	return NewAddress("127.0.0.1", 4222, identity.Address)
+func NewAddressForIdentity(identity identity.Identity, ipifyClient ipify.Client) *NatsAddress {
+	ipToBind, err := ipifyClient.GetOutboundIP()
+	if err != nil {
+		fmt.Errorf("failed to get outbound IP for NATS")
+	}
+	return NewAddress(ipToBind, 4222, identity.Address)
```

Comment:
NATS server address will be injected by variable similar like MYSTERIUM_API_URL.
And we have separate ticket for that MYST-27

Reply:
For now we may assume that NATS server and myst node runs on the same machine.